File: sgilt-core/src/main/java/net/franzka/sgilt/core/reservation/dto/FeedItemDto.java
package net.franzka.sgilt.core.reservation.dto;

import lombok.Builder;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * Représente un élément du feed d'une réservation (note ou document).
 * Le champ {@code type} détermine le type de l'item ("note" | "document").
 * Les champs non applicables au type sont null.
 */
@Builder
public record FeedItemDto(

        // ── Commun ──────────────────────────────────────────────────────────────
        String type,
        UUID id,
        String authorId,
        String authorName,
        String authorPhoto,
        String authorRole,
        LocalDateTime createdAt,
        String title,

        // ── Note ────────────────────────────────────────────────────────────────
        String content,
        Boolean isPersonal,
        Boolean isMessageInitial,

        // ── Document ─────────────────────────────────────────────────────────────
        String name,
        String fileType,
        String url,

        // ── Note système ──────────────────────────────────────────────────────────
        String generatedKey
) {}

File: sgilt-front/app/data/reservation/dto/FeedItemDto.ts
export interface FeedItemDto {
  type: 'note' | 'document'
  id: string
  authorId: string
  authorName: string
  authorPhoto?: string | null
  authorRole: 'client' | 'prestataire'
  createdAt: string
  title?: string | null

  // note
  content?: string | null
  isPersonal?: boolean | null
  isMessageInitial?: boolean | null

  // document
  name?: string | null
  fileType?: string | null
  url?: string | null
}
